pkg/config/config.go

```diff
@@ -439,20 +439,11 @@ func (c *Config) UpdateFromFile(path string) error {
 	t := new(tomlConfig)
 	t.fromConfig(c)
 
-	metadata, err := toml.Decode(string(data), t)
+	_, err = toml.Decode(string(data), t)
 	if err != nil {
 		return fmt.Errorf("unable to decode configuration %v: %v", path, err)
 	}
 
-	// If the default runtime `runc` provided via DefaultConfig() is not part
-	// of the configuration file, then we have to manually remove it because
-	// the user explicitly removed it
-	runtimesKey := []string{"crio", "runtime", "runtimes"}
-	if metadata.IsDefined(runtimesKey...) &&
-		!metadata.IsDefined(append(runtimesKey, defaultRuntime)...) {
-		delete(c.Runtimes, defaultRuntime)
-	}
```

Comment:
I have the feeling that we move back and forth with this change. What about an alternative solution for https://github.com/cri-o/cri-o/issues/3360 (which would not work any more if we remove those lines)?

Reply:
@saschagrunert with this fix I tested following scenario.

Now just like in case of https://github.com/cri-o/cri-o/issues/3360, user wants to add `crun` as their runtime of choice plus they also want to use it as their `default` runtime. With this change, they don't have to change anything in the crio.conf that ships with crio. All they have to do is add following drop in config at `/etc/crio/crio.conf.d/10-crun.conf`,

```bash
[crio.runtime]
  default_runtime = "crun"
  [crio.runtime.runtimes]
    [crio.runtime.runtimes.crun]
      runtime_path="/usr/local/bin/crun"
```
When you start crio, you will see following entries with `./crio config` command,

```toml
[crio.runtime.runtimes.crun]
runtime_path = "/usr/local/bin/crun"
runtime_type = ""
runtime_root = ""

[crio.runtime.runtimes.runc]
runtime_path = ""
runtime_type = "oci"
runtime_root = "/run/runc"
```
and, 
```toml
# default_runtime is the _name_ of the OCI runtime to be used as the default.
# The name is matched against the runtimes map below.
default_runtime = "crun"
```
The crio come up fine btw. 

The original intend of drop in config is to minimize the user interaction with the main crio.conf file. Having an independent drop in config allows user to focus only on the config they are interested in and not accidentally make unintentional changes to main crio.conf

In fact this PR helps the original reporter of https://github.com/cri-o/cri-o/issues/3360. Now if anyone wants to add a runtime and set it to default, all they have to do is just add a drop in config for that runtime and they are set.    

@saschagrunert @haircommander @mrunalp Let me know if I am making any sense.

Reply:
That's all fine from my side. The thing left open for me is how to start CRI-O if runc is not available on the host system?

Reply:
CRIO is explicitly adding a [default runtime](https://github.com/cri-o/cri-o/blob/master/pkg/config/config.go#L35) as `runc`. As long as we have that dependency user is not in position to remove `runc`. 

However, if you change those constants to point to `crun` and recompile,

```go
	defaultRuntime        = "crun"
	DefaultRuntimeType    = "oci"
	DefaultRuntimeRoot    = "/run/crun"
```
crio come up fine even if you remove `runc`. But now user cannot remove `crun` and add `runc` configuration. 

I think of it as an explicit declaration of dependency, user doesn't get to remove it.